oozie-to-airflow/tests/mappers/test_email_mapper.py

```diff
@@ -0,0 +1,124 @@
+
+        self.et = ET.ElementTree(email)
+
+    def test_create_mapper_no_jinja(self):
+        self._create_xml_no_jinja()
+        mapper = email_mapper.EmailMapper(oozie_node=self.et.getroot(),
+                                          task_id='test_id',
+                                          trigger_rule=TriggerRule.DUMMY)
+        # make sure everything is getting initialized correctly
+        self.assertEqual('test_id', mapper.task_id)
+        self.assertEqual(TriggerRule.DUMMY, mapper.trigger_rule)
+        self.assertEqual(self.et.getroot(), mapper.oozie_node)
+
+        self.assertEqual(['bob@apache.org', 'not.bob@apache.org'], mapper.to)
+        self.assertEqual(['john@apache.org', 'jeff@apache.org'], mapper.cc)
+        self.assertEqual(['secret@apache.org', 'secret2@apache.org'],
+                         mapper.bcc)
+        self.assertEqual('\'test_subject\'', mapper.subject)
+        self.assertEqual('\'this is a test body\'', mapper.body)
+        self.assertEqual('text/plain', mapper.content_type)
+        self.assertEqual(['/file/in/hdfs.txt'], mapper.attachment)
+
+    def test_create_mapper_jinja(self):
+        self._create_xml_jinja()
+        # test jinja templating
+        params = {'to': 'user@apache.org,user2@apache.org',
+                  'cc': 'bob@apache.org',
+                  'bcc': 'jeff@apache.org',
+                  'subject': 'test_subject',
+                  'content_type': 'text/plain',
+                  'body': 'this is a test body',
+                  'attachment': '/file/in/hdfs.txt'}
+
+        mapper = email_mapper.EmailMapper(oozie_node=self.et.getroot(),
+                                          task_id='test_id',
+                                          trigger_rule=TriggerRule.DUMMY,
+                                          params=params)
+        self.assertEqual('test_id', mapper.task_id)
+        self.assertEqual(TriggerRule.DUMMY, mapper.trigger_rule)
+        self.assertEqual(self.et.getroot(), mapper.oozie_node)
+        # First two are parameterized in oozie
+        self.assertEqual('\'{{ params.subject }}\'', mapper.subject)
+        self.assertEqual('\'{{ params.body }}\'', mapper.body)
+        # Not parameterized so must replace variable with actual
+        self.assertEqual(['user@apache.org', 'user2@apache.org'], mapper.to)
+        self.assertEqual(['bob@apache.org'], mapper.cc)
+        self.assertEqual(['jeff@apache.org'], mapper.bcc)
+        self.assertEqual('text/plain', mapper.content_type)
+        self.assertEqual(['/file/in/hdfs.txt'], mapper.attachment)
+
+    def test_convert_to_text(self):
+        self._create_xml_no_jinja()
+        mapper = email_mapper.EmailMapper(oozie_node=self.et.getroot(),
+                                          task_id='test_id',
+                                          trigger_rule=TriggerRule.DUMMY)
+        # Throws a syntax error if doesn't parse correctly
+        ast.parse(mapper.convert_to_text())
+
```

Comment:
Can we add a test case for defaults when fields are not specified. Maybe 1 test case for regular defaults and 1 test case for when a critical field is missing.